pkg/server/api/agent/v1/service.go

```diff
@@ -471,3 +618,206 @@ func (s *Service) getSelectorsFromAgentID(ctx context.Context, agentID string) (
 
 	return api.NodeSelectorsToProto(resp.Selectors)
 }
+
+func validateAttestAgentParams(params *agent.AttestAgentRequest_Params) error {
+	if params == nil {
+		return errors.New("missing params")
+	}
+	if params.Data == nil {
+		return errors.New("missing attestation data")
+	}
+	if params.Params == nil {
+		return errors.New("missing X509-SVID parameters")
+	}
+	if len(params.Params.Csr) == 0 {
+		return errors.New("missing CSR")
+	}
+	if params.Data.Type == "" {
+		return errors.New("missing attestation data type")
+	}
+	if params.Data.Payload == "" {
+		return errors.New("missing attestation data payload")
+	}
+	return nil
+}
+
+func (s *Service) isAttested(node *common.AttestedNode, baseSpiffeID string) bool {
+	if node != nil && node.SpiffeId == baseSpiffeID {
+		return true
+	}
+	return false
+}
+
+func (s *Service) isBanned(node *common.AttestedNode) bool {
+	if node != nil && nodeutil.IsAgentBanned(node) {
+		return true
+	}
+	return false
+}
```

Comment:
Doesn't use state on `Service`. How would you feel about making it standalone?
```suggestion
func isBanned(node *common.AttestedNode) bool {
	return node != nil && nodeutil.IsAgentBanned(node)
}
```

Reply:
I will make it standalone. I missed it stopped using state after the first refactor.